src/main/java/teammates/ui/controller/InstructorFeedbackResultsPageData.java

```diff
@@ -1636,10 +1649,15 @@ private boolean isResponseVisibleTo(FeedbackParticipantType participantType, Fee
         return moderationButtons;
     }
     
-    // TODO remove this entirely and use PageData method directly
     public String getProfilePictureLink(String studentEmail) {
-        return getStudentProfilePictureLink(StringHelper.encrypt(studentEmail),
-                                            StringHelper.encrypt(instructor.courseId));
+        if (!profilePictureLink.containsKey(studentEmail)) {
+            profilePictureLink.put(studentEmail, 
+                                   getStudentProfilePictureLink(StringHelper.encrypt(studentEmail),
+                                                                StringHelper.encrypt(instructor.courseId)));
```

Comment:
Should this override `PageData.getStudentProfilePictureLink()`?
Also, considering it's only used here, the method should be private.
As for the removed comment, should it be replaced with another comment targeting L1655 / 1656 directly or is the current behaviour correct and expected now?